src/sentry/similarity.py

```diff
@@ -0,0 +1,212 @@
+from __future__ import absolute_import
+
+import math
+import random
+
+
+def scale_to_total(value):
+    """\
+    Convert a mapping of distinct quantities to a mapping of proportions of the
+    total quantity.
+    """
+    total = float(sum(value.values()))
+    return {k: (v / total) for k, v in value.items()}
+
+
+def get_distance(target, other):
+    """\
+    Calculate the N-dimensional Euclidian between two mappings.
+
+    The mappings are used to represent sparse arrays -- if a key is not present
+    in both mappings, it's assumed to be 0 in the mapping where it is missing.
+    """
+    return math.sqrt(
+        sum(
+            (target.get(k, 0) - other.get(k, 0)) ** 2
+            for k in set(target) | set(other)
+        )
+    )
+
+
+def get_similarity(target, other):
```

Comment:
Was going to write a docstring for this and realized I wasn't really sure how to explain it coherently without the context from the `MinHashIndex` below. (I've gone back and forth as to whether it should actually be considered part of the class definition or not.)